os/board/rtl8721csm/src/component/common/api/wifi_interactive_mode.c

```diff
@@ -473,10 +481,23 @@ int8_t cmd_wifi_connect(trwifi_ap_config_s *ap_connect_config, void *arg)
 		password_len = ap_connect_config->passphrase_length;
 		semaphore = NULL;
 		break;
+	case TRWIFI_AUTH_WPA_AND_WPA2_PSK:
+		if (ap_connect_config->ap_crypto_type == TRWIFI_CRYPTO_TKIP_AND_AES) {
+			security_type = RTW_SECURITY_WPA_WPA2_MIXED_PSK;
+		} else if (ap_connect_config->ap_crypto_type == TRWIFI_CRYPTO_AES) {
+			security_type = RTW_SECURITY_WPA_WPA2_AES_PSK;
+		} else {
+			security_type = RTW_SECURITY_WPA_WPA2_TKIP_PSK;
+		}
+		password = ap_connect_config->passphrase;
+		ssid_len = strlen((const char *)ssid);
+		password_len = ap_connect_config->passphrase_length;
+		semaphore = NULL;

```

Comment:
I think these lines can be moved to above of switch

Reply:
ok can